kubespawner/reflector.py

```diff
@@ -150,16 +152,18 @@ def _list_and_update(self):
 
         Overwrites all current resource info.
         """
-        initial_resources = getattr(self.api, self.list_method_name)(
+        method = partial(getattr(self.api, self.list_method_name), _preload_content=False)
+        initial_resources = method(
             self.namespace,
             label_selector=self.label_selector,
             field_selector=self.field_selector,
             _request_timeout=self.request_timeout,
         )
         # This is an atomic operation on the dictionary!
-        self.resources = {p.metadata.name: p for p in initial_resources.items}
+        initial_resources = json.loads(initial_resources.read())
+        self.resources = {p["metadata"]["name"]: p for p in initial_resources["items"]}
```

Comment:
Technically this changes the value on the `resources` dict from an object to a dict so the help here should probably be updated as well:

https://github.com/jupyterhub/kubespawner/pull/424/files#diff-2f4d4efedf466e7bf4f8f083d0c722adR52

One thing I'm not sure about is how much this might impact out-of-tree reflectors that are expecting `self.resources` to return objects rather than dicts, i.e. this could be a breaking change for them. One could maybe mitigate that by wrapping the dicts in simple objects that pass through `__getattr__` to `__getitem__` but as we can see below with `resource_version` we'd also have to camel-case-ify some of the attributes, like [this](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/fcf9c2e92a450e175a9d53abbe6dbbb2936df096/jupyterhub/files/hub/jupyterhub_config.py#L39).

I'm not sure what the ABI contract on something like this is within the KubeSpawner. Would a release note be sufficient?